apps/etterlatte-behandling/src/main/kotlin/behandling/domain/ManueltOpphoer.kt

```diff
@@ -13,6 +14,7 @@ import java.util.*
 data class ManueltOpphoer(
     override val id: UUID,
     override val sak: Long,
+    override val sakType: SakType? = null,
```

Comment:
Trenger denne å være nullable egentlig? Kan vi ikke bare i flyway si at alle eksisterende saker er BARNEPENSJON? eller sette det som default for så å fjerne den senere?

Reply:
Tja - fant at opprett også bruker

asFoerstegangsbehandling og asManueltOpphoer osv.

Men - den bruker ikke verdien til opprettelse - siden det er bare oppretting av behandlingen.

Man kunne endre 

```kotlin
    private fun <T> ResultSet.valueOrNull(columnName: String, block: (ResultSet) -> T?): T? {
        return when (
            (1..this.metaData.columnCount).map {
                this.metaData.getColumnName(it)
            }.contains(columnName)
        ) {
            true -> block(this)
            false -> null
        }
    }
}
```

til setValueOrDefault og tar inn en default verdi - men den vil ikke brukes - og når det først er opprettet - så henter det fra sak senere. Så ved henting så join'er vi.

Er åpent for begge metoder egentlig - hva foretrekker folk?

Reply:
Men - hva skal man sette som default? P.t. så er det BP - men - det betyr at vi kommer til å sette BP på omstillingsstønad ved opprettelse - som vil da erstattes med riktig verdi fra sak senere - men føles feil ut

Reply:
Enig, ville trodd vi kunne migrert alle eksisterende til å være barnepensjon

Reply:
Eksisterende vil være barnepensjon ved henting allerede - dette er info vi har i basen.

Issuet er bare at dette konverteringsmetode brukes også når vi opprette en behandling - og der har vi ikke en verdi fra en join å sette.

Vi kan fjerne nullability uten å sette en falsk verdi hvis vi sette denne felt til ikke nullable fra frontend/andre backend (hvor det måtte komme fra) når det skal opprettes ny behandling.

Men det er å be om at vi blir sendt info vi burde allerede vite.

Reply:
Det er en annen mulighet - men liker den ikke helt.

Erstatt

```sql
INSERT INTO x (f1, sakId, f3, f4) VALUES (1,sakId, 2,3)
```

med noe a la

```sql
INSERT INTO x (f1, sakId, f3, f4, sakType) SELECT 1,sakId, 2,3, sakType FROM sak where sak.id == sakId
```

Reply:
Snakket med @sebassonav 

Ved lagring av status - fjernet retur - brukes ikke.

Ved opprettelse av manuelt opphør - istedenfor RETURNING * og heller hent ut på nytt

4 commits:

* Original
* Fjern returverdi fra lagreStatus
* Hent på nytt på Manuelt Opphør
* Fjern nullability på sakType i behandling

@sebassonav @madsop-nav @henrikg 

Merk - vi må nok også fikse de tingene som kaller de endrede endepunkter - de må også sender sakType i tillegg til sakId.

Men før vi gjøre det - vil jeg at dette reviewes på nytt :)